File: specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/examples-java/getPolicySetDefinitionAtManagementGroup.java

/** Samples for PolicySetDefinitions GetAtManagementGroup. */
public final class Main {
    /*
     * x-ms-original-file: specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/examples/
     * getPolicySetDefinitionAtManagementGroup.json
     */
    /**
     * Sample code: Retrieve a policy set definition at management group level.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void
        retrieveAPolicySetDefinitionAtManagementGroupLevel(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.genericResources().manager().policyClient().getPolicySetDefinitions()
            .getAtManagementGroupWithResponse("CostManagement", "MyManagementGroup", com.azure.core.util.Context.NONE);
    }
}

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2023-04-01/examples-js/getPolicySetDefinitionAtManagementGroup.js
const { PolicyClient } = require("@azure/arm-policy");
const { DefaultAzureCredential } = require("@azure/identity");
require("dotenv/config");

/**
 * This sample demonstrates how to This operation retrieves the policy set definition in the given management group with the given name.
 *
 * @summary This operation retrieves the policy set definition in the given management group with the given name.
 * x-ms-original-file: specification/resources/resource-manager/Microsoft.Authorization/stable/2023-04-01/examples/getPolicySetDefinitionAtManagementGroup.json
 */
async function retrieveAPolicySetDefinitionAtManagementGroupLevel() {
  const managementGroupId = "MyManagementGroup";
  const policySetDefinitionName = "CostManagement";
  const credential = new DefaultAzureCredential();
  const client = new PolicyClient(credential);
  const result = await client.policySetDefinitions.getAtManagementGroup(
    managementGroupId,
    policySetDefinitionName,
  );
  console.log(result);
}
